workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,31 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include <Arduino.h>
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+class XGraphics {
+protected:
+    XGraphics* _parent;
+public:
+    XGraphics();
+
+    virtual XColor getFGColor() const;
+
+    virtual XVector2<int16_t> getCanvasPivot() const;
+    virtual int16_t getCanvasWidth() const;
+    virtual int16_t getCanvasHeight() const;
+
+    virtual void  renderScanline(XRenderer* renderer, int16_t scanline, uint16_t bufferSize, uint16_t* buffer) {}
+    void renderScanlineRecursively(XRenderer* renderer, int16_t scanline, uint16_t bufferSize, uint16_t* buffer);
+    void renderCanvas(XRenderer* renderer);
```

Comment:
Why should this method belong to the base `XGraphics` and not to the `Canvas` itself